examples/static_map/device_side_example.cu

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <limits>
+
+#include <thrust/device_vector.h>
+#include <thrust/equal.h>
+#include <thrust/iterator/zip_iterator.h>
+#include <thrust/sequence.h>
+#include <thrust/transform.h>
+
+#include <cuco/static_map.cuh>
+
+/**
+ * @file device_side_example.cu
+ * @brief Demonstrates usage of the device side APIs for individual operations like insert/find.
+ *
+ * Individual operations like a single insert or find can be performed in device code via the
+ * static_map "device_view" types. Note that concurrent insert and find are not supported, and
+ * therefore there are separate view types for insert and find to help prevent undefined behavior.
+ *
+ * @note This example is for demonstration purposes only. It is not intended to show the most
+ * performant way to do the example algorithm.
+ *
+ */
+
+/**
+ * @brief Inserts keys that pass the specified predicated into the map
+ *
+ * @tparam Map Type of the map returned from static_map::get_mutable_device_view
+ * @tparam KeyIter Input iterator whose value_type convertible to Map::key_type
+ * @tparam ValueIter Input iterator whose value_type is convertible to Map::mapped_type
+ * @tparam Predicate Unary predicate
+ * @param[in] map_view View of the map into which inserts will be performed
+ * @param[in] key_begin The beginning of the range of keys to insert
+ * @param[in] value_begin The beginning of the range of values associated with each key to insert
+ * @param[in] num_keys The total number of keys and values
+ * @param[in] pred Unary predicate applied to each key. Only keys that pass the predicated will be
+ * inserted.
+ * @param[out] num_inserted The total number of keys successfully inserted
+ */
+template <typename Map, typename KeyIter, typename ValueIter, typename Predicate>
+__global__ void filtered_insert(Map map_view,
+                                KeyIter key_begin,
+                                ValueIter value_begin,
+                                std::size_t num_keys,
+                                Predicate pred,
+                                int* num_inserted)
+{
+  auto tid = threadIdx.x + blockIdx.x * blockDim.x;
```

Comment:
I think it is better to explicitly use int instead of auto because threadIdx.x is unsigned. If in any case it is involved in math that could product negative value, there could be a problem. Not in this kernel though.